doc/api/esm.md

```diff
@@ -798,6 +798,19 @@ indexes (e.g. `'./startup/index.js'`) must also be fully specified.
 This behavior matches how `import` behaves in browser environments, assuming a
 typically configured server.
 
+### No folder imports
```

Comment:
The above section mentions this on line 795 - 796

Do we need another section? If the above is not sufficient we may want to augment both sections so the level of detail and examples are consistent.

Reply:
Good catch, it just caught me off guard that folders with a `package.json` file in them are no longer supported so thought it's worth calling out explicitly.

Reply:
I think it could be reasonable to have two sections. if we were to do that we need to ensure that the information is not duplicated and that both sections have equivalent documentation

Reply:
Ping @thomheymann